### PR TITLE
expose a single file in the version api serializer

### LINKED ISSUE #17839: Expose a single File for each Version in the API
follow-up from #16465 - we've dropped multiple files per version at the model/database level and should do the same in the API.  
Note:
- it will be tricky to deploy without breaking anything. Might want to continue exposing files and file initially, deploy that, then following push start using file, then following push remove files.
- v3/v4 must continue to expose a list of files for backwards compatibility.

### PR DESCRIPTION
**Pull request by eviljeff**
fixes #17839 - exposes both `file` and `files` in v5 for now - we'll drop `files` once frontend is updated.

### CODE REVIEW

File: src/olympia/landfill/management/commands/fetch_prod_addons.py [hunk 1]
```diff
@@ -68,18 +68,15 @@ def _get_addons_from_page(self, page, params=None):
...
-
-        file_kw = {}
 
         try:
             file_kw = {
-                'hash': files[0]['hash'],
-                'status': amo.STATUS_CHOICES_API_LOOKUP[files[0]['status']],
-                'size': files[0]['size'],
-                'is_webextension': files[0]['is_webextension'],
+                'hash': version['file']['hash'],
```
Comment (bobsilverberg) [inline src/olympia/landfill/management/commands/fetch_prod_addons.py:74]: Nit: before there was a shortcut `files`, which made these statements a bit shorter. Would it be worthwhile to define a `file = version['file']` above and use that shortcut in all of these statements?

File: src/olympia/landfill/management/commands/fetch_prod_versions.py [hunk 2]
```diff
@@ -73,7 +73,7 @@ def _handle_version(self, data):
             print('Skipping %s (version already exists' % data['version'])
             return
 
-        files_data = data['files'][0]
+        files_data = data['file']
```
Comment (bobsilverberg) [inline src/olympia/landfill/management/commands/fetch_prod_versions.py:76]: Nit: should this be `file_data` now?

**Review by bobsilverberg — APPROVED** [review]
Looks good, nice cleanup @eviljeff!

I just added a couple of nits.

r+wc

File: src/olympia/landfill/management/commands/fetch_prod_addons.py (hunk 1 above)
Comment (eviljeff) [reply src/olympia/landfill/management/commands/fetch_prod_addons.py:74]: doesn't really make much difference - especially in landfill code (that's only executed once when a local dev setup is installed) but can do

**Pull request merged**